base/compiler/typelattice.jl

```diff
@@ -185,6 +193,14 @@ function ⊑(@nospecialize(a), @nospecialize(b))
         end
         return false
     end
+    if isa(a, PartialOpaque)
+        if isa(b, PartialOpaque)
+            (a.parent === b.parent && a.source === b.source) || return false
+            return (widenconst(a) <: widenconst(b)) &&
+                ⊑(a.env, b.env)
```

Comment:
```suggestion
                a.env_ts == b.env_ts
```
Not sure we need this check though (I think `a.parent === b.parent && a.source === b.source` implies this ?)

Reply:
It's possible to have the opaque closure creation in a loop, so you can end up having the same OC with two different envs. As for the actual change, I decided to leave the env as a type. There's just a rebase mistake that left in the old definition. Will fix.